packages/pangea-sdk/pangea/crypto/rsa.py

```diff
@@ -45,3 +57,73 @@ def public_key_to_pem(public_key: rsa.RSAPublicKey) -> bytes:
     return public_key.public_bytes(
         encoding=serialization.Encoding.PEM, format=serialization.PublicFormat.SubjectPublicKeyInfo
     )
+
+
+_AES_GCM_IV_SIZE = 12
+"""Standard nonce size for GCM."""
+
+_KEY_LENGTH = 32
+"""AES-256 key length in bytes."""
+
+
+def kem_decrypt(
+    private_key: rsa.RSAPrivateKey,
+    iv: bytes,
+    ciphertext: bytes,
+    symmetric_algorithm: str,
+    asymmetric_algorithm: str,
+    encrypted_salt: bytes,
+    password: str,
+    iteration_count: int,
+    hash_algorithm: str,
+) -> str:
+    if symmetric_algorithm.casefold() != SymmetricKeyEncryptionAlgorithm.AES_GCM_256.value.casefold():
+        raise NotImplementedError(f"Unsupported symmetric algorithm: {symmetric_algorithm}")
+
+    if asymmetric_algorithm != ExportEncryptionAlgorithm.RSA_NO_PADDING_4096_KEM:
+        raise NotImplementedError(f"Unsupported asymmetric algorithm: {asymmetric_algorithm}")
+
+    if hash_algorithm.casefold() != "SHA512".casefold():
+        raise NotImplementedError(f"Unsupported hash algorithm: {hash_algorithm}")
+
+    salt = private_key.decrypt(encrypted_salt, padding.PKCS1v15())
```

Comment:
Here is the issue. It should be decrypted with no padding. We should find a package that allow it or implement it somehow